jr-annotation-support/src/main/java/com/fasterxml/jackson/jr/annotationsupport/AnnotationBasedValueRWModifier.java

```diff
@@ -75,4 +97,115 @@ public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws
             return _valueType;
         }
     }
+
+    /**
+     * Serialize an enum using the {@link JsonValue} tagged method
+     */
+    private static class EnumJsonValueWriter implements ValueWriter {
+        private final Class<?> _valueType;
+        private final Method _jsonValueMethod;
+
+        private EnumJsonValueWriter(Class<?> _valueType, Method _jsonValueMethod) {
+            this._valueType = _valueType;
+            this._jsonValueMethod = _jsonValueMethod;
+        }
+
+        @Override
+        public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws IOException {
+            try {
+                context.writeValue(_jsonValueMethod.invoke(value));
+            } catch (Exception e) {
+                throw new IOException("Cannot call JsonValue method", e);
```

Comment:
I'd use Jackson jr specific exception here (`JSONObjectException`), not plain `IOException`.